parser/process_dataset.py

```
@@ -519,14 +519,14 @@ def convert_from_ftp(
-        ftp.login()  # Uses password: anonymous@
-        return ftp
-    except ftplib.all_errors as e:
-        logger.error(f'FTP login failed at {time.strftime("%c")}')
-        raise e
+    while True:
+        time.sleep(10)  # Delay to avoid rate limiting
+        try:
+            ftp = ftplib.FTP(ftp_ip)
+            ftp.login()  # Uses password: anonymous@
+            return ftp
+        except ftplib.all_errors as e:
```

Flake8 (7.3.0)

[error] local variable 'e' is assigned to but never used

(F841)